pennylane_lightning/core/src/utils/Util.hpp

```diff
@@ -24,7 +24,7 @@
 #include <numbers>
 #include <numeric> // transform_reduce
 #include <set>
-#include <type_traits> // is_same_v
+#include <type_traits> // integral, floating_point, is_same_v
 #include <vector>
```

Comment:
```suggestion
#include <vector>
#include <concepts> // integral, floating_point
```
reference: https://en.cppreference.com/w/cpp/header/concepts